store_bulk_ops.go

```
@@ -0,0 +1,686 @@
+package pgchainbs
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/filecoin-project/go-bs-postgres-chainnotated/lib/synccid"
+	"github.com/ipfs/go-cid"
+	pool "github.com/libp2p/go-buffer-pool"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/dustin/go-humanize"
+	"github.com/jackc/pgx/v4"
+	"golang.org/x/xerrors"
+)
+
+// Background initial state prefetch
+// Panic()s on early errors
```

Why panic rather than return an error? Even though it is run in a background goroutine it could return an error to that goroutine which could decide to log and ignore or panic itself